datalad/interface/add_archive_content.py

```diff
@@ -78,87 +77,107 @@
+            content of `example.zip` will be extracted under `example/`"""),
         strip_leading_dirs=Parameter(
             args=("--strip-leading-dirs",),
             action="store_true",
-            doc="""flag to move all files directories up, from how they were stored in an archive,
-                   if that one contained a number (possibly more than 1 down) single leading
-                   directories"""),
+            doc="""move all archive contents up from how they are stored in the
+             archive, if the archive contains a number (possibly more than 1
+             down) single leading directories"""),
         leading_dirs_depth=Parameter(
             args=("--leading-dirs-depth",),
             action="store",
             type=int,
-            doc="""maximal depth to strip leading directories to.  If not specified (None), no limit"""),
+            doc="""maximal depth to strip leading directories to.
+            If not specified (None), no limit"""),
         leading_dirs_consider=Parameter(
             args=("--leading-dirs-consider",),
             action="append",
-            doc="""regular expression(s) for directories to consider to strip away""",
+            doc="""regular expression(s) for directories to consider to strip
+            away""",
             constraints=EnsureStr() | EnsureNone(),
         ),
         use_current_dir=Parameter(
             args=("--use-current-dir",),
             action="store_true",
-            doc="""flag to extract archive under the current directory,  not the directory where archive is located.
-                   %s""" % _KEY_OPT_NOTE),
+            doc="""extract the archive under the current directory, not the
+             directory where the archive is located. This parameter is applied
+             automatically if [PY: `key=True` PY][CMD: --key CMD] was used."""),
         # TODO: add option to extract under archive's original directory. Currently would extract in curdir
         existing=Parameter(
             args=("--existing",),
             choices=('fail', 'overwrite', 'archive-suffix', 'numeric-suffix'),
             default="fail",
-            doc="""what operation to perform a file from archive tries to overwrite an existing
-             file with the same name.  'fail' (default) leads to RuntimeError exception.
-             'overwrite' silently replaces existing file.  'archive-suffix' instructs to add
-             a suffix (prefixed with a '-') matching archive name from which file gets extracted,
-             and if that one present, 'numeric-suffix' is in effect in addition, when incremental
-             numeric suffix (prefixed with a '.') is added until no name collision is longer detected"""
+            doc="""what operation to perform if a file from an archive tries to
+            overwrite an existing file with the same name.  'fail' (default)
+            leads to RuntimeError exception, 'overwrite' silently replaces
```

Comment:
If that is still true, it should be changed to yield an error result, not cause an exception.

Reply:
thx, I have implemented it